amethyst_renderer/src/pass/quad/interleaved.rs

```diff
@@ -252,48 +328,88 @@ impl SpriteBatch {
 
         let mut instance_data = Vec::<f32>::new();
         let mut num_instances = 0;
-        let num_sprites = self.sprites.len();
-
-        for (i, sprite) in self.sprites.iter().enumerate() {
-            // `unwrap`
-            let sprite_sheet = sprite_sheet_storage
-                .get(&sprite.render.sprite_sheet)
-                .expect(
-                    "Unreachable: Existence of sprite sheet checked when collecting the sprites",
-                );
+        let num_quads = self.quads.len();
 
+        for (i, quad) in self.quads.iter().enumerate() {
             let texture = tex_storage
-                .get(&sprite.texture)
+                .get(&quad.texture_handle())
                 .expect("Unable to get texture of sprite");
 
-            // Append sprite to instance data.
-            let sprite_data = &sprite_sheet.sprites[sprite.render.sprite_number];
+            let (dir_x, dir_y, pos, uv_left, uv_right, uv_top, uv_bottom) = match quad {
+                QuadDrawData::Sprite {
+                    render, transform, ..
+                } => {
+                    let sprite_sheet = sprite_sheet_storage
+                        .get(&render.sprite_sheet)
+                        .expect(
+                            "Unreachable: Existence of sprite sheet checked when collecting the sprites",
+                        );
 
-            let tex_coords = &sprite_data.tex_coords;
-            let (uv_left, uv_right) = if sprite.render.flip_horizontal {
-                (tex_coords.right, tex_coords.left)
-            } else {
-                (tex_coords.left, tex_coords.right)
-            };
-            let (uv_bottom, uv_top) = if sprite.render.flip_vertical {
-                (tex_coords.top, tex_coords.bottom)
-            } else {
-                (tex_coords.bottom, tex_coords.top)
+                    // Append sprite to instance data.
+                    let sprite_data = &sprite_sheet.sprites[render.sprite_number];
+
+                    let tex_coords = &sprite_data.tex_coords;
```

Comment:
Texture coordinates will be stored in a `Uv2D` component in the next pull requests. If its not present, it will default to the normal uv coords of a quad.